hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/entity/TopicImpl.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2022-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.consensus.impl.entity;
+
+import com.hedera.node.app.service.consensus.entity.Topic;
+import com.hedera.node.app.service.consensus.entity.TopicBuilder;
+import com.hedera.node.app.spi.key.HederaKey;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Objects;
+import java.util.Optional;
+
+/** An implementation of {@link Topic} */
+public record TopicImpl(
+        long topicNumber,
+        @Nullable HederaKey adminKey,
+        @Nullable HederaKey submitKey,
+        String memo,
+        long autoRenewAccountNumber,
+        long autoRenewSecs,
+        long expiry,
+        boolean deleted,
+        long sequenceNumber)
+        implements Topic {
+
+    @Override
+    public long shardNumber() {
```

Comment:
I'm pretty sure we'll never need to store shard or realm at the entity level; I think those are properties of the ledger as a whole.